### PR TITLE
feat: make data_source variable to use auotocomplete api and make auto select for required variable

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [ ] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description

SSIA

### CODE REVIEW

File: apps/web/src/services/dashboards/shared/dashboard-variables/DashboardVariablesDropdown.vue [hunk 1]
```diff
@@ -130,6 +133,12 @@ const state = reactive({
         }
         return result ?? [];
     }),
+    autocompleteApi: computed<ReturnType<typeof getCancellableFetcher>>(() => {
+        const api = state.variableProperty.options?.resource_key
+            ? SpaceConnector.client.addOns.autocomplete.distinct
+            : SpaceConnector.client.addOns.autocomplete.resource;
+        return getCancellableFetcher(api);
+    }),
```
Comment (yuda110) [inline apps/web/src/services/dashboards/shared/dashboard-variables/DashboardVariablesDropdown.vue:141]: praise: good!!

**Review by yuda110 — APPROVED** [review]
LGTM!!

File: apps/web/src/services/dashboards/shared/dashboard-variables/DashboardVariablesDropdown.vue (hunk 1 above)
Comment (piggggggggy) [reply apps/web/src/services/dashboards/shared/dashboard-variables/DashboardVariablesDropdown.vue:141]: I like it

**Review by piggggggggy — APPROVED** [review]
LGTM!

**Pull request merged**